nl.go

```diff
@@ -104,6 +104,8 @@ const (
 	DEVLINK_PORT_FUNCTION_ATTR_HW_ADDR = 1
 	DEVLINK_PORT_FN_ATTR_STATE         = 2
 	DEVLINK_PORT_FN_ATTR_OPSTATE       = 3
+	// attributes supported by genric NL 'mlxdevm'
+	MLXDEVM_PORT_FN_ATTR_TRUST = 4
```

Comment:
devlink and mlxdevm attr enums are usually in sync.
so if both families support, then you can use both sockets to send same msg.

where can i find the uapi of mlxdevm ?

upstream, 0x4 is used for "caps"[1]

[1] https://github.com/torvalds/linux/blob/8a696a29c6905594e4abf78eaafcb62165ac61f1/include/uapi/linux/devlink.h#L679

Reply:
Yes, upstream's definition is different.

According to the the comment in iproute2-mlnx_ofed_5_9/mlxdevm/include/uapi/linux/mlxdevm_netlink.h:
```
enum mlxdevm_port_fn_attr {
...
	MLXDEVM_PORT_FN_ATTR_TRUST_STATE =	4,	/* u8 */

	/* All upstream devlink port function attributes must be added before
	 * with the exact value as that of upstream without fail.
	 * All devm specific must start after MLXDEVM_PORT_FN_ATTR_EXT_START.
	 * Do not ever change the values. Only add at the end. Never in the
	 * middle.
	 */
	MLXDEVM_PORT_FUNCTION_ATTR_EXT_START =	160,

	MLXDEVM_PORT_FN_ATTR_EXT_CAP_ROCE,		/* u8 */
	MLXDEVM_PORT_FN_ATTR_EXT_CAP_UC_LIST,		/* u32 */
...
};
```

`MLXDEVM_PORT_FN_ATTR_TRUST_STATE` should be defined after  `MLXDEVM_PORT_FUNCTION_ATTR_EXT_START`.

Reply:
i think if mlxdevm kernel developers chose to go this path, then mlxdevm and devlink diverged

so i wonder if this lib should just be used for mlxdevm familiy.

@moshe010 @paravmellanox  thoughts ?

Reply:
@adrianchiris when mlxdevm is used than value 4 should be used for trust. And when devlink is used. 4 should do caps.

Reply:
thanks @paravmellanox currently i understand that devlink change for upstream trust mode is not happening, right? but if devlink supports trust in the future (not sure if cap is something that is relevant to mlxdevm) will this cause an issue. 

Just wondering why we didn't pick trust > 160 based on the comments in the file.. devlink would just fail for that
then..

Reply:
@paravmellanox after initial development I moved out to different project and when trust of upstream didnt happen, mlxdevm missed to do at different offset. :(

upstream is unlikely do do any trust knob. devlink will not cause any issue.
devlink will use offset 4 for the caps.
mlxdevm ill use offset 4 for trust, until it develops the caps at offset 4. Some new plumbing will be done to move the trust from offset 4 to higher value and to reclaim caps at 4 for mlxdevm.